qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -200,7 +201,10 @@ def _discover_backends_from_instance(self, instance: str) -> List[str]:
                     new_client = self._create_new_cloud_api_client(instance)
                     self._api_clients.update({instance: new_client})
                     self._active_api_client = new_client
-            return self._active_api_client.list_backends()
+                self._backends_list = self._active_api_client.list_backends()
+            if not self._backends_list:
+                self._backends_list = self._active_api_client.list_backends()
```

Comment:
if list_backends is going to be called anyways outside of the if-else, why not do:

```suggestion
            if not self._backends_list:
                self._backends_list = self._active_api_client.list_backends()
```